add MongoDB deployment and usage documentation

Rut-Vahab opened the pull request:
# PR Summary: MongoDB Deployment

- Deployed a simple MongoDB container for AgStream.
- Created scripts: `up.sh`, `down.sh`, `cleanup.sh`.
- Database is up, reachable from Jetson, and ready for use.

closes #68

Rut-Vahab committed:
add MongoDB deployment and usage documentation

Co-authored-by: ruti_vahab <rt80009000@gmail.com>
Co-authored-by: ruti_cohen <r0583283575@gmail.com>

Sarah5567 commented on database/docker-compose.yml:
```diff
@@ -0,0 +1,13 @@
+    container_name: agstream_mongo
+    restart: unless-stopped
+    ports:
+      - "27017:27017"
```
I think it might cause an error if two people try to connect at the same time, because the port is the same for everyone.

Rut-Vahab replied:
The comment isn’t correct in this case. We have a single MongoDB instance running on the Jetson, and multiple users or containers can connect to it at the same time without any problem. MongoDB supports multiple simultaneous connections, so sharing the same port does not cause errors.

Sarah5567 replied:
this setup works only if a single MongoDB instance runs on the Jetson.
if multiple users run their own containers, they must use different host ports
(e.g. 27018:27017, 27019:27017) to avoid port binding conflicts.

r83575 replied:
You're correct that multiple MongoDB instances on the same host would require different host ports.
However, in our setup we have only a single MongoDB instance running on the Jetson, and multiple users or containers simply connect to that single instance.
We validated this by testing concurrent connections from two separate containers using the same port (27017), and both were able to connect and operate simultaneously without any issue.

Sarah5567 commented on database/scripts/down.sh:
consider combining all the scripts into one file and using flags to determine which operation to perform.

Rut-Vahab replied:
That was the mission requirement to create scripts.
https://github.com/KamaTechOrg/AgStream/issues/68#issue-3531361111:~:text=set%20up)%2C%20create-,scripts,-for%20up/down

lyuzinmaxim commented on docs/mongodb_setup.md:
```diff
@@ -0,0 +1,34 @@
+
+## Connection
+
+Use the following connection string from Jetson or host:
```
jetson = host, I think you meant `from Jetson or container`

lyuzinmaxim commented on docs/mongodb_setup.md:
```diff
@@ -0,0 +1,34 @@
+## Notes
+
+* Run scripts from host, not inside another container.
+* `docker compose` must be installed on the host.
```
don't we have docker compose installed on host (jetson)?

lyuzinmaxim commented on env/arm/Dockerfile:
```diff
@@ -28,8 +28,14 @@ RUN apt-get update && apt-get install -y --no-install-recommends \
+    curl gnupg \
  && rm -rf /var/lib/apt/lists/*
 
+RUN curl -fsSL https://deb.nodesource.com/setup_20.x | bash - && \
```
Can we have a separate container for the database? Because the database shouldn't be connected to our ARM container

Rut-Vahab replied:
The database already runs in a separate container (agstream_mongo).
Our ARM container only connects to it — it doesn’t host or run MongoDB itself.
So the setup already follows the separation principle between the app and the database.

lyuzinmaxim replied:
Why do we install it then in the container?

lyuzinmaxim commented on docs/mongodb_setup.md:
can you please add information about how to push data to it, and how to check if the "write" is successful?

Also do we have an option to define the maximum storage space for the database? E.g. 100MB, so that we don't run out of memory on Jetson

Rut-Vahab replied:
we add to the docs how to push data and to check it .

About the memory limit - we don't have administrator permissions so we can't do that.

r83575 committed:
clarify MongoDB connection and host setup notes in docs/mongodb_setup.md

Co-authored-by: ruti_vahab <rt80009000@gmail.com>
Co-authored-by: ruti_cohen <r0583283575@gmail.com>

Rut-Vahab committed:
Expand MongoDB setup documentation with write instructions

Added instructions for writing data to MongoDB and verifying writes.

Rut-Vahab committed:
Updata Dockerfile

Rut-Vahab committed:
Update

Updated connection instructions to use 'docker exec' for MongoDB access and clarified script execution context.
